### ISSUE TITLE
AWS p4de.24xlarge instance type support

### ISSUE THREAD

**Issue by elgalu** [aws/karpenter-provider-aws#5455]
### Description

**What problem are you trying to solve?**
The p4d instance type is listed at https://karpenter.sh/docs/reference/instance-types/#p4d-family but p4de is missing
note p4d != p4de.

However the specs of p4de were added 1 year ago https://github.com/aws/karpenter-provider-aws/blob/6934ef58e8e13686d4f57c2d336dd165cbf3e759/pkg/providers/pricing/zz_generated.pricing_aws.go#L246

**How important is this feature to you?**
Karpenter is currently not scheduling our `p4de.24xlarge` pods so perhaps is not supported yet?

**Comment by jmdeal** [comment]
It looks like the `p4de.24xlarge` is still in preview, have you requested access? Are you running in one of the supported regions (us-east-1 or us-west-2)? As long as those are true it should be supported. If not could you attach logs and relevant objects (pods, nodeclass, nodepool, etc)?

**Comment by elgalu** [comment]
Yes, `p4de.24xlarge` is still in preview, we got access and we have are able to spin them up in our region via EC2 and also via EKS (without Karpenter). The problem arises with Karpenter. Let me check if we can bring the logs.

**Comment by jonathan-innis** [comment]
I think this problem is actually more general, we just need to support preview instance types. There's an issue that's tracking this here: #3585. Do you mind +1-ing that issue. We should think about raising that issue in priority to unblock for cases like this.

**Comment by njtran** [comment]
Closing this as duplicate of #3585 to track it there as the overarching issue
